### PR TITLE
feat(profiling): Update flamegraph thread selector

### PR DESCRIPTION
**Pull request by Zylphrex**
Update the thread selector on flamegraphs to use the new updated designs.

### CODE REVIEW

File: static/app/components/profiling/threadSelector.tsx [hunk 1]
```diff
@@ -1,226 +1,87 @@
...
+        name: profile.name,
+        duration: profile.duration,
+        index: i,
+      }))
+      .sort(compareProfiles)
+      .map(item => ({label: item.name, value: item.index}));
+  }, [profileGroup]);
+
+  const handleChange = useCallback(
+    opt => {
```
Comment (JonasBa) [inline static/app/components/profiling/threadSelector.tsx:33]: Nit, but the opt here is any type. Can you add the type from the select component or something? Maybe just type handleChange: CompactSelectProps[onChange] or something

File: static/app/components/profiling/threadSelector.tsx [hunk 2]
```diff
@@ -1,226 +1,87 @@
...
+      if (defined(opt)) {
```
Comment (JonasBa) [inline static/app/components/profiling/threadSelector.tsx:34]: when can this be undefined? 🤔

**Review by JonasBa — APPROVED** [review]
Minor comments, but LGTM. Nice to see we can reuse that component 🔥

File: static/app/components/profiling/threadSelector.tsx (hunk 2 above)
Comment (Zylphrex) [reply static/app/components/profiling/threadSelector.tsx:34]: honestly not sure, this was just part of the type from the compact select

**Pull request merged**